packages/d/dotnet/xmake.lua

```diff
@@ -0,0 +1,59 @@
+            add_versions("7.0.202", "1cc74b279cffe42f757b3adeeeec87e351f0fb806016167ffb6b3842d4d6bcd5")
+        else
+            set_urls("https://download.visualstudio.microsoft.com/download/pr/c7213e17-4955-4fae-a405-1b7101086daf/71963248a7de64c4b00219a8199e7429/dotnet-sdk-7.0.202-win-x86.zip")
+            add_versions("7.0.202", "8541bc89a12dac90ba1d6a7c1b6875b9314fc8af04602210be07daa16cfd5e9a")
+        end
+    elseif is_host("linux") then
+        if os.arch() == "x86_64" then
+            set_urls("https://download.visualstudio.microsoft.com/download/pr/bda88810-e1a6-4cf0-8139-7fd7fe7b2c7a/7a9ffa3e12e5f1c3d8b640e326c1eb14/dotnet-sdk-7.0.202-linux-x64.tar.gz")
+            add_versions("7.0.202", "405f15e437582be260460f48eda9dfe613fd87b2557667f20d6ecfa34b09c221")
+        elseif os.arch() == "arm64" then
+            set_urls("https://download.visualstudio.microsoft.com/download/pr/c1fd11b0-186a-4aa1-a578-bb1b6613886e/b67e1c9d6d90b1c99b23935273921fa1/dotnet-sdk-7.0.202-linux-arm64.tar.gz")
+            add_versions("7.0.202", "5a5d29f3f7515f6df888047eeb9571a2fa06b728a97e30536b1473d2abce0b32")
+        end
+    elseif is_host("macosx") then
+        if os.arch() == "x86_64" then
+            set_urls("https://download.visualstudio.microsoft.com/download/pr/d3fe9043-0ff4-4584-aacf-1ad41c47407b/7b84ed341359488cd0de21de1b4df6d0/dotnet-sdk-7.0.202-osx-x64.tar.gz")
+            add_versions("7.0.202", "1531bfa7bf384f8b5c433d87af170f7839b130ea9c20b617aaf41742a13eefa9")
+        elseif os.arch() == "arm64" then
+            set_urls("https://download.visualstudio.microsoft.com/download/pr/4402413a-ef81-4732-a0c0-38f11694b226/e205b8bf48d95902a6dbe1c3cccca272/dotnet-sdk-7.0.202-osx-arm64.tar.gz")
+            add_versions("7.0.202", "7f8201d42d31d6985288d0e31a32aa3b1dff40fceb8cb202993c0b72883bcebb")
+        end
+    end
+
+    on_install("windows|x86", "windows|x64", "linux|x64", "linux|arm64", "macosx|x86_64", "macosx|arm64", function (package)
+
+        local out_path = "packs"
+        if package:is_plat("windows") then
+            out_path = path.join(out_path, "Microsoft.NETCore.App.Host.win-" .. (package:is_arch("x64") and "x64" or "x86"), "7.0.4", "runtimes", "win-" .. (package:is_arch("x64") and "x64" or "x86"), "native")
+            
+            os.cp(path.join(out_path, "nethost.dll"), package:installdir("bin"))
+            os.cp(path.join(out_path, "nethost.lib"), package:installdir("lib"))
+        elseif package:is_plat("linux") then
+            out_path = path.join(out_path, "Microsoft.NETCore.App.Host.linux-" .. (package:is_arch("arm64") and "arm64" or "x64"), "7.0.4", "runtimes", "linux-" .. (package:is_arch("arm64") and "arm64" or "x64"), "native")
+
+            os.cp(path.join(out_path, "*.so"), package:installdir("bin"))
+            os.cp(path.join(out_path, "*.a"), package:installdir("lib"))
```

Comment:
We should distinguish between static/shared, otherwise the linker may link to the wrong library.

and so and dylib should be in lib directory instead of bin.

```
if package:config("shared") then
            os.cp(path.join(out_path, "*.so"), package:installdir("lib"))
else
            os.cp(path.join(out_path, "*.a"), package:installdir("lib"))
end

Reply:
There is no static version, the .a is used to reference the .so

Reply:
Only the windows dll library may rely on .lib as an import library.

The dynamic libraries of linux and macos do not depend on the .a static library at all. There is no relationship between them.

I looked at libnethost.so and libnethost.a for linux and they both contain separate symbolic definitions with no cross-reference.

```
ruki@88b79cd6fd38:/mnt/dotnet-sdk-7.0.202-linux-x64/packs/Microsoft.NETCore.App.Host.linux-x64/7.0.4/runtimes/linux-x64/native$ nm libnethost.a | grep get_hostfxr_path
0000000000000000 T get_hostfxr_path
ruki@88b79cd6fd38:/mnt/dotnet-sdk-7.0.202-linux-x64/packs/Microsoft.NETCore.App.Host.linux-x64/7.0.4/runtimes/linux-x64/native$ nm libnethost.so | grep get_hostfxr_path
00000000000140c0 T get_hostfxr_path
```

So they should be completely separate dynamic and static libraries, and only one of them can be linked and used at the same time. Otherwise there may be link symbol conflicts, as well as runtime loading conflicts.